### PR TITLE
Move slicing and plotting into mixin classes

### LINKED ISSUE #33: Use the NDData Slicing mixin for NDCube
This https://github.com/astropy/astropy/blob/master/astropy/nddata/mixins/ndslicing.py looks like it could reduce the complexity of our slicing implementation. The biggest divergences look like our handling of `uncertainty` and the extra coords handling.

### PR DESCRIPTION
**Pull request by Cadair**
This continues my refactoring quest. It moves the plotting code into a mixin and also closes #33 by subclassing that mixin class to pull in all our slicing code. (Note there is a minor difference in the handling of `uncertainty`).

### CODE REVIEW

File: ndcube/mixins/ndslicing.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                    except IndexError:
+                        pass
+                    except TypeError:
+                        pass
+                new_extra_coords_dict = self._extra_coords_to_input_format(new_extra_coords,
+                                                                           missing_axis)
+
+        return new_extra_coords_dict
+
+    def _extra_coords_to_input_format(self, extra_coords, missing_axis):
```
Comment (Cadair) [inline ndcube/mixins/ndslicing.py]: I think this will need a PR to IRISPy

File: ndcube/mixins/ndslicing.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    def __getitem__(self, item):
+        """
+        Override to exclude None indices.
+        """
+        if item is None or (isinstance(item, tuple) and None in item):
+            raise IndexError("None indices not supported")
+
+        return super().__getitem__(item)
+
+    def _slice(self, item):
```
Comment (Cadair) [inline ndcube/mixins/ndslicing.py]: add a decent docstring here, to describe why this exists.

File: ndcube/mixins/ndslicing.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    def _slice_wcs(self, item):
+        """
+        Override this so it does not interfere.
+        """
+        return None
+
+    def __getitem__(self, item):
+        """
+        Override to exclude None indices.
```
Comment (Cadair) [inline ndcube/mixins/ndslicing.py]: Change this so that it says it creates a new instance of the super class.

**Comment by DanRyanIrish** [comment]
Looks good.  But some more comments on the interplay between inheritence of mixins, etc.

File: ndcube/ndcube.py [hunk 4]
```diff
@@ -91,14 +86,51 @@ def __init__(self, data, wcs, uncertainty=None, mask=None, meta=None,
...
+            self._extra_coords_wcs_axis = self._format_input_extra_coords_to_extra_coords_wcs_axis(
                 extra_coords, self.missing_axis, data.shape)
         else:
             self._extra_coords_wcs_axis = None
         # Initialize NDCube.
         super(NDCube, self).__init__(data, wcs=wcs, uncertainty=uncertainty, mask=mask,
                                      meta=meta, unit=unit, copy=copy, **kwargs)
 
+    @staticmethod
+    def _format_input_extra_coords_to_extra_coords_wcs_axis(extra_coords, missing_axis,
```
Comment (Cadair) [inline ndcube/ndcube.py]: move to utils

**Pull request merged**